hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/test/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtilTest.java

```diff
@@ -0,0 +1,88 @@
+
+    result = GraphBuilderUtil.isSameStructuredTrace(cachedTrace, null);
+    Assertions.assertFalse(result);
+  }
+
+  @Test
+  public void testSameTraceForTenantAndTraceCondition() {
+    // different tenant id
+    StructuredTrace cachedTrace = mock(StructuredTrace.class);
+    when(cachedTrace.getCustomerId()).thenReturn("__defaultTenant");
+    when(cachedTrace.getTraceId()).thenReturn(ByteBuffer.wrap("2ebbc19b6428510f".getBytes()));
+
+    StructuredTrace underTestTrace = mock(StructuredTrace.class);
+    when(underTestTrace.getCustomerId()).thenReturn("__defaultTenantUnderTest");
+    when(underTestTrace.getTraceId()).thenReturn(ByteBuffer.wrap("2ebbc19b6428510f".getBytes()));
+
+    boolean result = GraphBuilderUtil.isSameStructuredTrace(cachedTrace, underTestTrace);
+    Assertions.assertFalse(result);
+
+    // different trace ids
+    cachedTrace = mock(StructuredTrace.class);
+    when(cachedTrace.getCustomerId()).thenReturn("__defaultTenant");
+    when(cachedTrace.getTraceId()).thenReturn(ByteBuffer.wrap("2ebbc19b6428510f".getBytes()));
+
+    underTestTrace = mock(StructuredTrace.class);
+    when(underTestTrace.getCustomerId()).thenReturn("__defaultTenant");
+    when(underTestTrace.getTraceId()).thenReturn(ByteBuffer.wrap("2ebbc19b6428511f".getBytes()));
+
+    result = GraphBuilderUtil.isSameStructuredTrace(cachedTrace, underTestTrace);
+    Assertions.assertFalse(result);
+  }
+
+  @Test
+  public void testSameTraceForSizeCondition() {
+    Entity entity = mock(Entity.class);
+    Event parent = mock(Event.class);
+    Event child = mock(Event.class);
+    Edge eventEdge = mock(Edge.class);
+
+    // same size
+    StructuredTrace cachedTrace = mock(StructuredTrace.class);
+    when(cachedTrace.getCustomerId()).thenReturn("__defaultTenant");
+    when(cachedTrace.getTraceId()).thenReturn(ByteBuffer.wrap("2ebbc19b6428510f".getBytes()));
+    when(cachedTrace.getEntityList()).thenReturn(List.of(entity));
+    when(cachedTrace.getEventList()).thenReturn(List.of(parent, child));
+    when(cachedTrace.getEntityEdgeList()).thenReturn(List.of());
+    when(cachedTrace.getEntityEventEdgeList()).thenReturn(List.of());
+    when(cachedTrace.getEventEdgeList()).thenReturn(List.of(eventEdge));
+
+    StructuredTrace underTestTrace = mock(StructuredTrace.class);
+    when(underTestTrace.getCustomerId()).thenReturn("__defaultTenant");
+    when(underTestTrace.getTraceId()).thenReturn(ByteBuffer.wrap("2ebbc19b6428510f".getBytes()));
+    when(underTestTrace.getEntityList()).thenReturn(List.of(entity));
+    when(underTestTrace.getEventList()).thenReturn(List.of(parent, child));
+    when(underTestTrace.getEntityEdgeList()).thenReturn(List.of());
+    when(underTestTrace.getEntityEventEdgeList()).thenReturn(List.of());
+    when(underTestTrace.getEventEdgeList()).thenReturn(List.of(eventEdge));
+
+    boolean result = GraphBuilderUtil.isSameStructuredTrace(cachedTrace, underTestTrace);
+    Assertions.assertTrue(result);
```

Comment:
Test the negative case as well?

Reply:
The remains are covered with the other two tests, and this I have created as an end to cover up all. But, if you want, I can add one to the list item.

Reply:
I think it can be taken up subsequently